pkg/chrootarchive/archive.go

```diff
@@ -35,13 +48,17 @@ func Untar(tarArchive io.Reader, dest string, options *archive.TarOptions) error
 		}
 	}
 
-	decompressedArchive, err := archive.DecompressStream(tarArchive)
-	if err != nil {
-		return err
+	r := ioutil.NopCloser(tarArchive)
```

Comment:
decompress already would wrapping an `io.Reader`, and if needed I'm handling closing that `io.ReadCloser` that it returns.

As a workflow, this is to say that if someone passed an `os.File` to `archive.Untar()` as an `io.Reader`, then I would imagine the `os.File` to be closed by the caller. That is to say, this NopCloser doesn't matter. Regardless, it is something that _could_ be cleaned up, that the `invokeUnpack()` calls for an `io.ReadCloser` but does not actually need to `Close()` it ...